src/monocooler.ts

```diff
@@ -0,0 +1,555 @@
+import {
+  BigInt,
+  BigDecimal,
+  Bytes,
+  ethereum,
+  Address,
+  Int8,
+} from "@graphprotocol/graph-ts";
+import {
+  CollateralAdded,
+  CollateralWithdrawn,
+  Borrow,
+  Repay,
+  Liquidated,
+  InterestRateSet,
+  LtvOracleSet,
+  LiquidationsPausedSet,
+  BorrowPausedSet,
+  TreasuryBorrowerSet,
+  MonoCooler,
+} from "../generated/MonoCooler/MonoCooler";
+import {
+  MonoCoolerAccount,
+  MonoCoolerGlobalState,
+  MonoCoolerAccountSnapshot,
+  MonoCoolerGlobalSnapshot,
+  MonoCoolerActivity,
+  MonoCoolerLtvOracleChange,
+  MonoCoolerLoanOrigination,
+  MonoCoolerLiquidation,
+} from "../generated/schema";
+
+// Helper: Get LTV values from the contract (raw BigInt WAD values)
+function getLtvValues(contractAddress: Address): BigInt[] {
+  let contract = MonoCooler.bind(contractAddress);
+  let ltvResult = contract.try_loanToValues();
+  
+  if (!ltvResult.reverted) {
+    let maxOriginationLtv = ltvResult.value.getMaxOriginationLtv();
+    let liquidationLtv = ltvResult.value.getLiquidationLtv();
+    return [maxOriginationLtv, liquidationLtv];
+  } else {
+    // Fallback values if contract call fails (example values in WAD)
+    return [BigInt.fromString("3000000000000000000000"), BigInt.fromString("3300000000000000000000")]; // 3000 and 3300 USDS/gOHM
+  }
+}
+
+// Helper: Get or create the global state singleton
+function getOrCreateGlobalState(
+  contractAddress: Address,
+  timestamp: BigInt
+): MonoCoolerGlobalState {
+  let id = "singleton";
+  let state = MonoCoolerGlobalState.load(id);
+  if (!state) {
+    state = new MonoCoolerGlobalState(id);
+    
+    // Load initial values from the contract instead of defaulting to zero
+    let contract = MonoCooler.bind(contractAddress);
+    
+    // Load contract state
+    let totalCollateralResult = contract.try_totalCollateral();
```

Comment:
If any of these were to revert during indexing, what would it indicate? Can they even revert?